### PR TITLE
feat(src/pt): New source: Animes CX

### LINKED ISSUE #1831: Extension Request: Animes CX
### Source name

Animes CX

### Source link

https://animes.cx/

### Language

Portuguese - Brazil / PT-BR

### Other details

+18/NSFW = Yes

It has Anime and Dramas

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a title with source name.
- [X] I have checked that the extension does not already exist on the [website extensions list](https://aniyomi.org/extensions/) or the app.
- [X] I have checked that the extension does not already exist by searching the [GitHub repository](https://github.com/aniyomiorg/aniyomi-extensions/) and verified it does not appear in the code base.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by Claudemirovsky**
Closes #1831
^ it's already closed by the bot so it doesn't count ok

Note: The source is pretty cursed, so the behaviour of this extension is kinda schrodinger-like when trying to watch movies: sometimes it works, sometimes it evaporates half of the solar system. It's a **fun and entertaining** built-in gacha game, NOT A BUG!!!!

Checklist:

- [ ] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [ ] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio
- [x] Have removed `web_hi_res_512.png` when adding a new extension

### CODE REVIEW

File: src/pt/animescx/src/eu/kanade/tachiyomi/animeextension/pt/animescx/AnimesCX.kt [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+                .awaitSuccess()
+                .use(::searchAnimeByIdParse)
+        } else {
+            super.getSearchAnime(page, query, filters)
+        }
+    }
+
+    private fun searchAnimeByIdParse(response: Response): AnimesPage {
+        val details = animeDetailsParse(response.asJsoup())
+            .apply { setUrlWithoutDomain(response.request.url.toString()) }
```
Comment (Secozzi) [inline src/pt/animescx/src/eu/kanade/tachiyomi/animeextension/pt/animescx/AnimesCX.kt:104]: Add `initialized = true` as well

File: src/pt/animescx/src/eu/kanade/tachiyomi/animeextension/pt/animescx/AnimesCX.kt [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+        return AnimesPage(listOf(details), false)
+    }
+
+    override fun searchAnimeRequest(page: Int, query: String, filters: AnimeFilterList) =
+        GET("$baseUrl/page/$page/?s=$query")
+
+    override fun searchAnimeSelector() = "article.rl_episodios:has(.rl_AnimeIndexImg)"
+
+    override fun searchAnimeFromElement(element: Element) = SAnime.create().apply {
+        element.selectFirst("a")!!.run {
```
Comment (Secozzi) [inline src/pt/animescx/src/eu/kanade/tachiyomi/animeextension/pt/animescx/AnimesCX.kt:114]:
```suggestion
        with(element.selectFirst("a")!!) {
```

File: src/pt/animescx/src/eu/kanade/tachiyomi/animeextension/pt/animescx/AnimesCX.kt (hunk 1 above)
Comment (Claudemirovsky) [reply src/pt/animescx/src/eu/kanade/tachiyomi/animeextension/pt/animescx/AnimesCX.kt:104]: Now that you said, we'll have to make this change in most extensions with a similar function for the URL intent parser too.

File: src/pt/animescx/src/eu/kanade/tachiyomi/animeextension/pt/animescx/AnimesCX.kt [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+    }
+
+    private fun searchAnimeByIdParse(response: Response): AnimesPage {
+        val details = animeDetailsParse(response.asJsoup())
+            .apply { setUrlWithoutDomain(response.request.url.toString()) }
+        return AnimesPage(listOf(details), false)
+    }
+
+    override fun searchAnimeRequest(page: Int, query: String, filters: AnimeFilterList) =
+        GET("$baseUrl/page/$page/?s=$query")
```
Comment (Secozzi) [inline src/pt/animescx/src/eu/kanade/tachiyomi/animeextension/pt/animescx/AnimesCX.kt:109]:
```suggestion
        GET("$baseUrl/page/$page/?s=$query", headers)
```

File: src/pt/animescx/src/eu/kanade/tachiyomi/animeextension/pt/animescx/AnimesCX.kt [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+                    }
+                    "Google Drive" -> {
+                        GDRIVE_REGEX.find(it.url)?.groupValues?.get(0)
+                            ?.let { gdriveExtractor.videosFromUrl(it, "GDrive - $quality") }
+                            .orEmpty()
+                    }
+                    else -> emptyList()
+                }
+            }
+        }
```
Comment (Secozzi) [inline src/pt/animescx/src/eu/kanade/tachiyomi/animeextension/pt/animescx/AnimesCX.kt:214]:
```suggestion
        }.sort()
```

**Review by Secozzi — APPROVED** [review]

**Comment by hollowshiroyuki** [comment]
yay gacha game in my Aniyomi™️ extension :3

File: src/pt/animescx/src/eu/kanade/tachiyomi/animeextension/pt/animescx/AnimesCX.kt [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+
+    override fun latestUpdatesNextPageSelector(): String? {
+        throw UnsupportedOperationException()
+    }
+
+    // =============================== Search ===============================
+    override suspend fun getSearchAnime(page: Int, query: String, filters: AnimeFilterList): AnimesPage {
+        return if (query.startsWith(PREFIX_SEARCH)) { // URL intent handler
+            val path = query.removePrefix(PREFIX_SEARCH)
+            client.newCall(GET("$baseUrl/$path"))
```
Comment (Secozzi) [inline src/pt/animescx/src/eu/kanade/tachiyomi/animeextension/pt/animescx/AnimesCX.kt:94]:
```suggestion
            client.newCall(GET("$baseUrl/$path", headers))
```

**Review by jmir1 — APPROVED** [review]

**Pull request merged**